sdk/python/feast/feature_store.py

```diff
@@ -22,13 +27,64 @@ class FeatureStore:
     """
 
     def __init__(
-        self, config_path: Optional[str], config: Optional[Config],
+        self, config_path: Optional[str] = None, config: Optional[Config] = None,
     ):
-        if config_path is None or config is None:
-            raise Exception("You cannot specify both config_path and config")
+        if config_path is not None and config is not None:
+            raise ValueError("You cannot specify both config_path and config")
         if config is not None:
             self.config = config
         elif config_path is not None:
             self.config = Config.from_path(config_path)
         else:
             self.config = Config()
+
+    def _pull_table(
+        self, feature_view: FeatureView, start_date: datetime, end_date: datetime
+    ) -> Optional[pyarrow.Table]:
+        if feature_view.inputs.table_ref is None:
+            raise NotImplementedError(
+                "Ingestion is not yet implemented for query-based sources."
+            )
+
+        # if we have mapped fields, use the original field names in the call to the offline store
+        event_timestamp_column = feature_view.inputs.event_timestamp_column
+        fields = (
+            [entity.name for entity in feature_view.entities]
+            + [feature.name for feature in feature_view.features]
+            + [feature_view.inputs.event_timestamp_column]
+        )
+        if feature_view.inputs.field_mapping is not None:
+            reverse_field_mapping = {
+                v: k for k, v in feature_view.inputs.field_mapping.items()
+            }
+            event_timestamp_column = (
+                reverse_field_mapping[event_timestamp_column]
+                if event_timestamp_column in reverse_field_mapping.keys()
+                else event_timestamp_column
+            )
+            fields = [
+                reverse_field_mapping[col]
+                if col in reverse_field_mapping.keys()
+                else col
+                for col in fields
+            ]
+
+        table = BigQueryOfflineStore.pull_table(
```

Comment:
One thing that we should remember to implement is to only return the latest feature data from the source table for each key.

Reply:
Right, easiest place to do this seems to be when ingesting into the online store but I wasn't grabbing the created_timestamp_column from BigQuery so I'm doing that now